### ISSUE TITLE
Improve multi-parent span visualization

### ISSUE THREAD

**Issue by andremarianiello** [jaegertracing/jaeger-ui#299]
## Requirement - what kind of business use case are you trying to solve?

TL;DR - I want to user Jeager UI to visualize traces with spans that have multiple span references in a way that makes those multiple references useful.

Actual reason why I have spans with multiple references:

I have a pipeline with multiple stages: A, B, C, etc. I have multiple elements flowing through this pipeline: 1, 2, 3, etc. Each pipeline stage has a span representing the beginning and ending execution of that stage. Each element has a span when it enters the pipeline until it leaves. I also create a span for when element n is processes by stage X. Logically, (in my opinion) the span C2 (element 2 is processed by stage C) is a child of span C, and follows from B2, the previous pipeline stage. Luckily, I can represent these relationships using the jaeger opentracing client. The visualization I get for these multi-parent spans, however, depends on the order that I set the references, but the visualization I want might not match that.

## Problem - what in Jaeger blocks you from solving the requirement?

Jaeger UI simply chooses a span reference to be the parent in the tree visualization. There is no visual relationship between a span and its non-primary span references.

## Proposal - what do you suggest to solve the problem or improve the existing situation?

I'm not a designer, so I don't really have a great idea for this, but I will share what I have considered:
* Display multiple copies of the child span, one beneath each parent span (this DAG->Tree conversion can lead to exponential blowup in the pathological case, so we will have to be careful how it is done). Provide a way to jump between copies of the child span in the visualization to aid in navigating
* Have a single copy of the child span in the visualization, but on the parent spans provide a button that moves all child spans underneath it. This essentially pivots the view of the trace. For example, if A has children B and C, and B and C have child D, clicking B will make D appear below B (as if B was the primary reference), and clicking C will make D appear below C (as if C was the primary reference). It would also be nice in this case to click on D and select a reference to move it under
* Steal an existing visualization for DAGs

**Comment by yurishkuro** [comment]
Are you specifically interested in using the Gantt Chart? Because if we simply draw a trace as DAG then the multi-tenancy situation is trivial to display, but the DAG will not provide any timing/duration information.

**Comment by andremarianiello** [comment]
I still want the timing and metadata information that the current visualization provides.

**Comment by objectiser** [comment]
Possibly could be achieved by showing dependency links (thin black lines maybe with arrows) between the child cell and the multiple parent (reference) cells? Might need to ensure the child is always shown below all of the parents.

**Comment by yurishkuro** [comment]
Someone needs to do visual prototype (mock is fine). Aside from DAG view, I don't yet see how multiple references can be shown in the Gantt chart.

**Comment by yurishkuro** [comment]
This issue #358 is just to surface reference information in span details (not in graphical form).
